MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -1474,54 +1504,141 @@ EnggDiffractionViewQtGUI::splitFittingDirectory(std::string &selectedfPath) {
   return splitBaseName;
 }
 
+void MantidQt::CustomInterfaces::EnggDiffractionViewQtGUI::enableMultiRun(
+    std::string firstRun, std::string lastRun) {
+
+  bool firstDig = isDigit(firstRun);
+  bool lastDig = isDigit(lastRun);
+
+  std::vector<std::string> RunNumberVec;
+  if (firstDig && lastDig) {
+    int firstNum = std::stoi(firstRun);
+    int lastNum = std::stoi(lastRun);
+
+    if ((lastNum - firstNum) > 200) {
+      userWarning(
+          "Please try again",
+          "The specified run number range is "
+          "far to big, please try a smaller range of consecutive run numbers.");
+    }
+
+    else if (firstNum <= lastNum) {
+
+      for (int i = firstNum; i <= lastNum; i++) {
+        RunNumberVec.push_back(std::to_string(i));
+      }
+
+      // if given a single run number instead
+      for (size_t i = 0; i < RunNumberVec.size(); i++) {
+        updateFittingDirVec(m_focusDir, RunNumberVec[i], true);
+      }
+      int diff = (lastNum - firstNum) + 1;
+      auto global_vec_size = m_fitting_runno_dir_vec.size();
+      if (size_t(diff) == global_vec_size) {
+
+        addRunNoItem(RunNumberVec, true);
+
+        emit setBank();
+      }
+    } else {
+      userWarning("Invalid Run Number", "One or more run file not found "
+                                        "from the specified range of runs."
+                                        "Please try again");
```

Comment:
Also, when I enter "194747" or any of the numbers that are in principle available, I don't get any error message but the combo box and the "Run number" list on the right of the plot are both grayed out.
Selecting files directly with "browse" seems to work well though. I get the right bank numbers in the combo box and the run number is shown in the "Run number" list.